invenio_requests/assets/semantic-ui/js/invenio_requests/timelineCommentEditor/state/actions.js

```diff
@@ -32,11 +38,27 @@ export const submitComment = (content, format) => {
 
     try {
       const response = await config.requestsApi.submitComment(payload);
-      dispatch({
+      const queryParams = new QueryBuilder(timelineState.queryParams);
+      const currentCommentsLength = timelineState.data.hits.hits.length;
+      const shouldGoToNextPage = currentCommentsLength + 1 > queryParams.size;
+
+      if (shouldGoToNextPage) {
+        const nextPage = queryParams.page + 1;
+        const newQueryParams = queryParams.setPage(nextPage).get();
+
+        dispatch({ type: SETTING_PARAMS, payload: newQueryParams });
```

Comment:
As discussed, hide internal implementations to the outside world and expose clear APIs, eg.:
```suggestion
        dispatch({ type: CHANGE_PAGE, payload: {currentPage: 3} });
```